### PR TITLE
Add action container info

### PR DESCRIPTION
**Pull request by mudra-octopus**
https://trello.com/c/3Y2yvRXZ/8-documentation

### CODE REVIEW

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+title: Execution container for workers
```
Comment (emilol) [inline docs/deployment-process/execution-containers-for-workers/index.md:2]:
```suggestion
title: Execution containers for workers
```

To be consistent

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+description: Setting an execution container for workers in steps to define your project's deployment process.
```
Comment (emilol) [inline docs/deployment-process/execution-containers-for-workers/index.md:3]: I have to admit, I'm a little lost reading this. I'm finding the compound sentence hard to parse because of the number of nouns (and therefore clauses) in the sentence. Does this say the same thing?

```suggestion
description: How to set a step in your deployment process to run inside a container.
```

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+position: 15
+---
+
+For a [step](/docs/deployment-process/steps/index.md) running on a [worker](docs/infrastructure/workers/index.md) or on [Octopus server](docs/infrastructure/workers/built-in-worker.md) that you define in your [deployment processes](/docs/deployment-process/index.md), you can select a container for the step's execution. 
+
+## Setup a container 
+
+- Set a [feed](/docs/packaging-applications/package-repositories/docker-registries/index.md) in Octopus Deploy for a Docker registry.  or link to https://octopus.com/blog/build-a-real-world-docker-cicd-pipeline#add-docker-hub-as-an-external-feed
+- Add a project and define deployment process.
```
Comment (emilol) [inline docs/deployment-process/execution-containers-for-workers/index.md:12]:
```suggestion
- Add a project and define a deployment process. 
```

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+For a [step](/docs/deployment-process/steps/index.md) running on a [worker](docs/infrastructure/workers/index.md) or on [Octopus server](docs/infrastructure/workers/built-in-worker.md) that you define in your [deployment processes](/docs/deployment-process/index.md), you can select a container for the step's execution. 
+
+## Setup a container 
+
+- Set a [feed](/docs/packaging-applications/package-repositories/docker-registries/index.md) in Octopus Deploy for a Docker registry.  or link to https://octopus.com/blog/build-a-real-world-docker-cicd-pipeline#add-docker-hub-as-an-external-feed
+- Add a project and define deployment process. 
+- Select a worker pool. 
+- Select to run on a container for worker. 
+- Choose the previously added container registry. 
+- Enter an image name if already known. (e.g. octopusdeploy/step-execution:0.1.1-ubuntu.18.04)
```
Comment (emilol) [inline docs/deployment-process/execution-containers-for-workers/index.md:16]:
```suggestion
- Enter an image name if already known. (e.g. !docker-image <octopusdeploy/worker-tools:ubuntu.18.04>)
```

This will automatically keep the docs up to date when we add new images. [See docs](https://octopusdeploy.github.io/OctoStyle/markdown#referencing-docker-images)

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+- Alternatively use the wizard to chose appropriate image.
```
Comment (emilol) [inline docs/deployment-process/execution-containers-for-workers/index.md:17]:
```suggestion
- Alternatively use the wizard to chose appropriate image. 
```

References a wizard we don't have at the moment.

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+---
+title: Execution containers for workers
+description: How to set a step in your deployment process to run inside a container.
+position: 15
+---
+
+For a [step](/docs/deployment-process/steps/index.md) running on a [worker](/docs/infrastructure/workers/index.md) or on [Octopus server](/docs/infrastructure/workers/built-in-worker.md) that you define in your [deployment processes](/docs/deployment-process/index.md), you can select a container for the step's execution. 
+
+## Minimum requirements for your custom image to run as a worker
```
Comment (emilol) [inline docs/deployment-process/execution-containers-for-workers/index.md:9]:
```suggestion
## Minimum requirements for your step to run in a container
```

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+The worker will need Docker installed.
```
Comment (emilol) [inline docs/deployment-process/execution-containers-for-workers/index.md:10]: We should link here to the requirements for a worker, and then say "all of those, plus docker"

Comment (ankithkonda) [reply docs/deployment-process/execution-containers-for-workers/index.md:10]: @emilol I assume those requirements will be on DockerHub right?

Comment (emilol) [reply docs/deployment-process/execution-containers-for-workers/index.md:10]: I don't think so. The minimum requirements for a worker relate to the host machine that `docker run` will get called from, not the image itself. You can check out the WorkerTools repo for the readme structure we've landed on for DockerHub

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+## How to use execution containers for workers 
+
+- Configure a [feed](/docs/packaging-applications/package-repositories/docker-registries/index.md) in Octopus Deploy for a Docker registry.
+  - [Add Docker Hub as an external feed](https://octopus.com/blog/build-a-real-world-docker-cicd-pipeline#add-docker-hub-as-an-external-feed)
+- Add a project and define a deployment process.
+- Set the `Execution Location` for your step to `Run on a worker`.
+- In `Container Image` select `Runs on a worker inside a container`.
+- Choose the previously added container registry.
+- Enter an image name if already known. (e.g. !docker-image <octopusdeploy/worker-tools:ubuntu.18.04>).
```
Comment (emilol) [inline docs/deployment-process/execution-containers-for-workers/index.md:25]:
```suggestion
- Enter the name of the image (aka execution container) you want your step to run in. (e.g. !docker-image <octopusdeploy/worker-tools:ubuntu.18.04>).
```

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+:::hint
+The same process can be repeated for a runbook.
+:::
+
+![](images/selector.png "width=500")
+
+## First deployment on a docker container
+Your first deployment on a Docker container might take a while as it will download all the dependencies on the docker image. These dependencies will be installed as a part of deployment process execution automatically. Alternatively, you can pre-pull the desired docker image before your first deployment to avoid any delays.
+
+## How to construct an image name
```
Comment (emilol) [inline docs/deployment-process/execution-containers-for-workers/index.md:38]:
```suggestion
## What docker image should I use?
```

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+:::
+
+![](images/selector.png "width=500")
+
+## First deployment on a docker container
+Your first deployment on a Docker container might take a while as it will download all the dependencies on the docker image. These dependencies will be installed as a part of deployment process execution automatically. Alternatively, you can pre-pull the desired docker image before your first deployment to avoid any delays.
+
+## What docker image should I use?
+
+You can find our recommended images on DockerHub [octopusdeploy/worker-tools](https://hub.docker.com/r/octopusdeploy/worker-tools)
```
Comment (emilol) [inline docs/deployment-process/execution-containers-for-workers/index.md:40]:
```suggestion

- You can use any image that meets the minimum tooling requirements to run your chosen step. 
- You can use our recommended images
- You can build your own based on the recommended images.

## The worker-tools docker images 

We provide recommended images on DockerHub [octopusdeploy/worker-tools](https://hub.docker.com/r/octopusdeploy/worker-tools) that include common tools used for octopus steps. 

```

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+## The worker-tools docker images 
+
+We provide recommended images on DockerHub [octopusdeploy/worker-tools](https://hub.docker.com/r/octopusdeploy/worker-tools) that include common tools used for octopus steps. 
+
+
+Pick an image that is suitable for your needs based on:
+- OS + distribution
+
+The image also needs to supports the version formats:
```
Comment (emilol) [inline docs/deployment-process/execution-containers-for-workers/index.md:53]:
```suggestion
The worker tools images have multiple release tracks, based on how you would like to receive updates as they're pushed to docker hub.
```

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+- Click **Create release & deploy**.
+
+:::hint
+The same process can be repeated for a runbook.
+:::
+
+![](images/selector.png "width=500")
+
+## First deployment on a docker container
+Your first deployment on a Docker container might take a while as it will download all the dependencies on the docker image. These dependencies will be installed as a part of deployment process execution automatically. Alternatively, you can pre-pull the desired docker image before your first deployment to avoid any delays.
```
Comment (emilol) [inline docs/deployment-process/execution-containers-for-workers/index.md:36]:
```suggestion
:::hint
Pre-pulling your chosen image will save you time during deployments.
:::

Your first deployment in a Docker container might take a while as it will download all the dependencies on the docker image. These dependencies will be installed as a part of deployment process execution automatically. Alternatively, you can pre-pull the desired docker image before your first deployment to avoid any delays.
```

**Comment by wordlee** [comment]
Please let me know when this is no longer a draft (or click ready for review), and I'll take a look.

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+Pre-pulling your chosen image will save you time during deployments.
+:::
+
+Your first deployment in a Docker container might take a while as it will download all the dependencies on the docker image. These dependencies will be installed as a part of deployment process execution automatically. Alternatively, you can pre-pull the desired docker image before your first deployment to avoid any delays.
+
+## What docker image should I use?
+
+
+- You can use any image that meets the minimum tooling requirements to run your chosen step. 
+- You can use our recommended images
```
Comment (MarkSiedle) [inline docs/deployment-process/execution-containers-for-workers/index.md:46]:
```suggestion
- You can use our recommended images (see below).
```

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+- Major.Minor
+- Major.Minor.Patch
+
+If the operating system on your worker is Linux, you can use any of the following images: 
+
+- octopusdeploy/worker-tools:ubuntu.18.04
+- octopusdeploy/worker-tools:1-ubuntu.18.04
+- octopusdeploy/worker-tools:1.0-ubuntu.18.04
+- octopusdeploy/worker-tools:1.0.1-ubuntu.18.04
+
```
Comment (MarkSiedle) [inline docs/deployment-process/execution-containers-for-workers/index.md:68]: And if my operation system on my worker in Windows? Do I just use `octopusdeploy/worker-tools` ?

Comment (ankithkonda) [reply docs/deployment-process/execution-containers-for-workers/index.md:68]: Ah no, so there are windows workers, but not on docker hub just yet. Once they're up I'll add in the details here :)

Comment (emilol) [reply docs/deployment-process/execution-containers-for-workers/index.md:68]: @ankithkonda this information feels like it would be better on the docker hub page for worker-tools. Mind if I move it there?

Comment (ankithkonda) [reply docs/deployment-process/execution-containers-for-workers/index.md:68]: Sure! that sounds good

Comment (emilol) [reply docs/deployment-process/execution-containers-for-workers/index.md:68]: [done](https://github.com/OctopusDeploy/WorkerTools/blob/master/README.dockerhub.md)

**Comment by ankithkonda** [comment]
@wordlee this is ready for review now :)

**Comment by emilol** [comment]
@wordlee would you mind putting your eye over the [dockerhub docs](https://github.com/OctopusDeploy/WorkerTools/blob/master/README.dockerhub.md) we link to them from this page too? We'll be uploading them manually once this merges.

File: docs/deployment-process/execution-containers-for-workers/index.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
+---
+title: Execution containers for workers
+description: How to set a step in your deployment/runbook process to run inside a container.
+position: 15
+---
+
+For a [step](/docs/deployment-process/steps/index.md) running on a [worker](docs/infrastructure/workers/index.md) or on the [Octopus Server](docs/infrastructure/workers/built-in-worker.md) that you define in your [deployment processes](/docs/deployment-process/index.md), you can select a container for the step's execution.
```
Comment (wordlee) [inline docs/deployment-process/execution-containers-for-workers/index.md:7]: Final tweak to simplify the sentence structure and make the point a bit clearer:
```suggestion
For a [step](/docs/deployment-process/steps/index.md) running on a [worker](docs/infrastructure/workers/index.md) or on the [Octopus Server](docs/infrastructure/workers/built-in-worker.md), you can select a container for the step's execution.
```

**Comment by ankithkonda** [comment]
merging docs now, OctopusDeploy/OctopusDeploy#5719 has been merged.
